### ISSUE TITLE
v14.0 matrix variable RANK ERROR

### ISSUE THREAD

**Issue by norberturkiewicz** [the-carlisle-group/Acre-Desktop#235]
@PhilLast Should this work in 14.0?

 ]acRE.createproject c:\temp\testprod test
#.test.test
#.test.test.Federal←⊃(2010 'sadf' 'sadf' 0 )(2011 'one' 'two' 1)
#.test.test.Federal
2010  sadf  sadf  0
2011  one   two   1
]setchanged #.test.test.Federal
 #.test.test.Federal 

* Command Execution Failed: RANK ERROR

**Comment by PhilLast** [comment]
I don't know what's causing your error but I don't even get that far. I get
```
      ]createproject acreprojects/testprod test
* Command Execution Failed: FILE NAME ERROR
```
It seems that the _File System Object_ function _CreateFolder_ requires that the parent folder has to exist so it can't create _testprod/APLSource_ until _testprod_ exists. I don't know why this only shows up now unless no-one has actually created a project in 14.0 for a very long time. I guess that before I replaced all the file stuff with the new system functions I had an iterative call to _CreateFolder_ and forgot to do the same for the emulator.

Nor do I understand why you get _#.test.test_ created when you specify _test_ in the argument. If I do this in later interpreters it creates _#.test_ as I'd expect.

**Comment by aplteam** [comment]
I think Norberts report is lacking two initial lines:

'test'⎕ns''
)cs test

Only then does it make sense

**Comment by PhilLast** [comment]
Yes. With that it correctly attempts to set the project in #.test.test in v18.0 but still it doesn't get far enough to hit Norbert's error in v14.0.

**Comment by PhilLast** [comment]
I guess no-one has created a project in _any_ version recently because the recent fix to the JSON emulator to correct a temporary Dyalog bug in importing JSON introduces an acre bug in _exporting_ it.

**Comment by aplteam** [comment]
There should be a test case for that.

**Comment by PhilLast** [comment]
Actually there was. Anything that called _CreateProject_ would fail because in trying to write the config file it was misinterpreting the config namespace as an array. The JSON emulator was only written to handle objects so we ended up with a strange blend of JSON object and APLAN array.
```
[⋄{⋄"CaseCode": false,⋄"Dependencies": "",⋄"KeepHistory": false,⋄"ProjectSpace": "#.test",⋄"StartUp": ""⋄}⋄]
```
that obviously won't evaluate under any encoding.

My mistake. I seem to have omitted the running of the tests prior to committing the very minor change and haven't changed anything since until yesterday. This was fixed by commit yesterday and reported in #236
